File: problemas/ad-hoc/1129/1129.jl
using Printf

while true
  n = parse(Int32, readline())

  if n == 0
    break
  end

  for i in 1:n
    opcao = -1
    corretos = 0
    
    cores = [parse(Int32, x) for x in split(readline())]
    for k in 1:5
      if cores[k] <= 127
        opcao = k
        corretos += 1
      end
    end

    if corretos == 1
      @printf "%c\n" (Char(65 + opcao - 1))
    else
      @printf "*\n"
    end
  end
end

File: problemas/ad-hoc/1129/1129.scala
object Main {
  def main(args: Array[String]): Unit = {
    var n = io.StdIn.readLine().toInt
    while (n != 0){
      for (i <- 1 to n){
        val cores = io.StdIn.readLine().split(" ").map(_.toInt)
  
        var opcao = -1
        var corretos = 0
        for (k <- 0 to 4){
          if (cores(k) <= 127){
            opcao = k
            corretos += 1
          }
        }
  
        if(corretos == 1){
          printf("%c\n", ('A'.toInt + opcao).toChar)
        }else{
          printf("*\n")
        }
      }
      
      n = io.StdIn.readLine().toInt
    }
  }
}
